### PR TITLE
Remove pound keywords token kinds

### PR DESCRIPTION
**Pull request by ahoppen**
These are no longer being used, so we can clean up the list of token kinds a little further.

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

File: gyb_syntax_support/Token.py [hunk 1]
```diff
@@ -227,17 +227,6 @@ def macro_name(self):
...
-    PoundKeyword('PoundKeyPath', 'keyPath', text='#keyPath'),
-    PoundKeyword('PoundLine', 'line', text='#line'),
-    PoundKeyword('PoundSelector', 'selector', text='#selector'),
-    PoundKeyword('PoundFile', 'file', text='#file'),
-    PoundKeyword('PoundFileID', 'fileID', text='#fileID'),
-    PoundKeyword('PoundFilePath', 'filePath', text='#filePath'),
-    PoundKeyword('PoundColumn', 'column', text='#column'),
-    PoundKeyword('PoundFunction', 'function', text='#function'),
-    PoundKeyword('PoundDsohandle', 'dsohandle', text='#dsohandle'),
     PoundKeyword('PoundAssert', 'assert', text='#assert'),
```
Comment (DougGregor) [inline gyb_syntax_support/Token.py:230]: `#assert` should be removable as well.

**Review by DougGregor — APPROVED** [review]
Thank you!

File: gyb_syntax_support/Token.py (hunk 1 above)
Comment (ahoppen) [reply gyb_syntax_support/Token.py:230]: We are still using it here

https://github.com/apple/swift-syntax/blob/3ca3c0bc15124009119d490e9aba998332f5dff5/Sources/SwiftParser/Statements.swift#L119-L120

Comment (DougGregor) [reply gyb_syntax_support/Token.py:230]: Alright, let's get rid of that later. It can be parsed as a freestanding macro expression.

**Pull request merged**